tests/common.go

```diff
@@ -8691,3 +8702,71 @@ func UpdateCloudCredentialOwnership(cloudCredentialName string, cloudCredentialU
 	}
 	return nil
 }
+
+func ValidateCRMigration(pods *v1.PodList, appData *asyncdr.AppData) error {
+	pods_created_len := len(pods.Items)
+	log.InfoD("Num of Pods on source: %v", pods_created_len)
+	sourceClusterConfigPath, err := GetSourceClusterConfigPath()
+	if err != nil {
+		return err
+	}
+	err = asyncdr.ValidateCRD(appData.ExpectedCrdList, sourceClusterConfigPath)
+	if err != nil {
+		return err
+	}
+	options := scheduler.ScheduleOptions{Namespace: appData.Ns}
+	var emptyCtx = &scheduler.Context{
+		UID:             "",
+		ScheduleOptions: options,
+		App: &spec.AppSpec{
+			Key:      "",
+			SpecList: []interface{}{},
+		}}
+	log.InfoD("Create cluster pair between source and destination clusters")
+	ScheduleValidateClusterPair(emptyCtx, false, true, defaultClusterPairDir, false)
+	migName := migrationKey + time.Now().Format("15h03m05s")
+	mig, err := asyncdr.CreateMigration(migName, appData.Ns, asyncdr.DefaultClusterPairName, appData.Ns, &includeVolumesFlag, &includeResourcesFlag, &startApplicationsFlag)
+	if err != nil {
+		return err
+	}
+	migrationList = append(migrationList, mig)
+	err = asyncdr.WaitForMigration(migrationList)
+	if err != nil {
+		return fmt.Errorf("Migration failed")
+	}
+	// Sleeping here, as apps deploys one by one, which takes time to collect all pods
+	time.Sleep(5 * time.Minute)
+	SetDestinationKubeConfig()
+	pods_migrated, err := core.Instance().GetPods(appData.Ns, nil)
+	if err != nil {
+		return err
+	}
+	pods_migrated_len := len(pods_migrated.Items)
+	log.InfoD("Num of Pods on dest: %v", pods_migrated_len)
+	if pods_created_len != pods_migrated_len {
+		return fmt.Errorf("Pods migration failed as %v pods found on source and %v on destination", pods_created_len, pods_migrated_len)
+	}
+	destClusterConfigPath, err := GetDestinationClusterConfigPath()
+	if err != nil {
+		return err
+	}
+	err = asyncdr.ValidateCRD(appData.ExpectedCrdList, destClusterConfigPath)
```

Comment:
Should not we be validating the CR itself?

Reply:
This will verify CRD seperately and we CR validation will be pods on source and destination.

Reply:
Its better to do the CR validation in fact, pods are just an aftereffect of the CR being present. If it cant happen in this PR, its ok but ideally we should check for the CR migration itself.

Reply:
Sure, I will create a backlog and assign it to me for this.